java/configuration/src/test/java/sleeper/configuration/properties/table/InMemoryTableProperties.java

```diff
@@ -62,11 +65,18 @@ public static TablePropertiesStore getStoreReturningExactInstances(List<TablePro
     public TableProperties loadProperties(TableIdentity tableId) {
         return Optional.ofNullable(propertiesByTableId.get(tableId.getTableUniqueId()))
                 .map(this::copyIfSet)
-                .orElseThrow();
+                .orElseThrow(() -> TableNotFoundException.withTableIdentity(tableId));
     }
 
     @Override
     public void saveProperties(TableProperties tableProperties) {
+        Optional<TableProperties> existingTableWithNewName = propertiesByTableId.values().stream()
+                .filter(table -> !tableProperties.get(TABLE_ID).equals(table.get(TABLE_ID)))
+                .filter(table -> tableProperties.get(TABLE_NAME).equals(table.get(TABLE_NAME)))
+                .findFirst();
+        if (existingTableWithNewName.isPresent()) {
+            throw new TableAlreadyExistsException(existingTableWithNewName.get().getId());
```

Comment:
The underlying storage can't really check this. There's no equivalent behaviour in the S3 store. In the DynamoDB/S3 implementation this check happens in the table index, so I think this should be in the in-memory table index?

Reply:
Thanks, i've moved this check into InMemoryTableIndex